vassal-app/src/main/java/VASSAL/build/BadDataReport.java

```diff
@@ -156,15 +199,55 @@ public void setReportable(boolean reportable) {
     this.reportable = reportable;
   }
 
+  /**
+   * Build a single line printable message for this Bad Data Report
+   * @return message
+   */
   public String getMessage() {
-    return message;
+    final StringBuilder sb = new StringBuilder();
+    if (owner != null) {
+      sb.append(getDescription(owner)).append(":");
+    }
+    if (data != null) {
+      sb.append(" ").append(Resources.getString("Audit.source", data));
+    }
+    sb.append(" ").append(Resources.getString("Audit.error", message));
+    return sb.toString();
+  }
+
+  private String getDescription(Auditable owner) {
+    return owner.getComponentName() + " " + owner.getComponentTypeName();
   }
 
   public Throwable getCause() {
     return cause;
   }
 
+  private void setCause(Throwable cause) {
+    this.cause = cause;
+    if (cause instanceof ExpressionException) {
+      auditTrail = ((ExpressionException) cause).getAuditTrail();
+      owner =  ((ExpressionException) cause).getOwner();
+    }
+  }
+
   public String getData() {
     return data;
   }
+
+  public AuditTrail getAuditTrail() {
+    return auditTrail;
+  }
+
+  public String getAuditReport() {
+    return auditTrail == null ? "" : auditTrail.toString();
+  }
+
+  /**
+   * Return the full audit trail if it exi
```

Comment:
Lost a couple of characters off the end of the comment on line 247.

Reply:
Thanks Jonathon.